arangod/Aql/IResearchViewExecutor.tpp

```diff
@@ -434,9 +598,13 @@ void IndexReadBuffer<ValueType, copySorted>::pushSortedValue(
     TRI_ASSERT(!sortContext.descScore() ||
                threshold <= _scoreBuffer[_rows.front() * _numScoreRegisters]);
 #endif
-    threshold = _scoreBuffer[_rows.front() * _numScoreRegisters];
+    if (!scores.empty()) {
+      threshold = _scoreBuffer[_rows.front() * _numScoreRegisters];
```

Comment:
What about making _scoreBuffer 1-based indexed.
I mean always have 0 element which will be used in case _numScoreRegisters == 0

Reply:
Well we can just always allocate at least one element in `_scoreBuffer`

Reply:
Yes, my suggestion about that, and you're right, we don't need to make it 1-based.